### PR TITLE
OM-229

### PR DESCRIPTION
**Pull request by Citron-mohammad-Rafi**

### CODE REVIEW

**Comment by Citron-mohammad-Rafi** [comment]
@MHigel Please review this PR if it is fine then merge it.

**Review by MHigel — CHANGES_REQUESTED** [review]
@Citron-mohammad-Rafi we need to move the `assignSubsidiary` method in the `beforeInsert` method to be the first method that's invoked so that the other methods that have logic relating to values in the `Subsidiary__c` field are able to correctly process the data point.

**Comment by Citron-mohammad-Rafi** [comment]
@MHigel I have make changes according to comment please check. And merge it if everything is fine.

**Review by MHigel — CHANGES_REQUESTED** [review]

**Review by MHigel — APPROVED** [review]

File: force-app/main/default/classes/ServiceTerritoryAssignmentTest.cls [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        for(Account ac : listOfAccounts){
+            ac.Subsidiary__c ='CA';
+            ac.Service_Territory__c = null;
+        }
+        insert listOfAccounts;
+    }
+ 
+    @isTest
+    static void testSetServiceTerritoryIns() {
+        test.startTest();
```
[the remark this answers is not in the document]
Comment (Citron-mohammad-Rafi) [reply force-app/main/default/classes/ServiceTerritoryAssignmentTest.cls:18]: @MHigel sure I will do it.

**Pull request merged**